x-pack/plugins/apm/ftr_e2e/cypress/e2e/read_only_user/mobile/generate_data.ts

```diff
@@ -0,0 +1,221 @@
+      cityName: 'Beijing',
+      continentName: 'Asia',
+      countryIsoCode: 'CN',
+      countryName: 'China',
+      regionIsoCode: 'CN-BJ',
+      regionName: 'Beijing',
+      location: { coordinates: [116.3861, 39.9143], type: 'Point' },
+    })
+    .setNetworkConnection({ type: 'wifi' });
+
+  const galaxy7 = apm
+    .mobileApp({
+      name: 'synth-android',
+      environment: 'production',
+      agentName: 'android/java',
+    })
+    .mobileDevice({ serviceVersion: SERVICE_VERSIONS[1] })
+    .deviceInfo({
+      manufacturer: 'Samsung',
+      modelIdentifier: 'SM-G930F',
+      modelName: 'Galaxy S7',
+    })
+    .osInfo({
+      osType: 'android',
+      osVersion: '10',
+      osFull: 'Android 10, API level 29, BUILD A022MUBU2AUD1',
+      runtimeVersion: '2.1.0',
+    })
+    .setGeoInfo({
+      clientIp: '223.72.43.22',
+      cityName: 'Beijing',
+      continentName: 'Asia',
+      countryIsoCode: 'CN',
+      countryName: 'China',
+      regionIsoCode: 'CN-BJ',
+      regionName: 'Beijing',
+      location: { coordinates: [116.3861, 39.9143], type: 'Point' },
+    })
+    .setNetworkConnection({
+      type: 'cell',
+      subType: 'edge',
+      carrierName: 'M1 Limited',
+      carrierMNC: '03',
+      carrierICC: 'SG',
+      carrierMCC: '525',
+    });
+
+  const huaweiP2 = apm
+    .mobileApp({
+      name: 'synth-android',
+      environment: 'production',
+      agentName: 'android/java',
+    })
+    .mobileDevice({ serviceVersion: SERVICE_VERSIONS[2] })
+    .deviceInfo({
+      manufacturer: 'Huawei',
+      modelIdentifier: 'HUAWEI P2-0000',
+      modelName: 'HuaweiP2',
+    })
+    .osInfo({
+      osType: 'android',
+      osVersion: '10',
+      osFull: 'Android 10, API level 29, BUILD A022MUBU2AUD1',
+      runtimeVersion: '2.1.0',
+    })
+    .setGeoInfo({
+      clientIp: '20.24.184.101',
+      cityName: 'Singapore',
+      continentName: 'Asia',
+      countryIsoCode: 'SG',
+      countryName: 'Singapore',
+      location: { coordinates: [103.8554, 1.3036], type: 'Point' },
+    })
+    .setNetworkConnection({
+      type: 'cell',
+      subType: 'edge',
+      carrierName: 'Osaka Gas Business Create Co., Ltd.',
+      carrierMNC: '17',
+      carrierICC: 'JP',
+      carrierMCC: '440',
+    });
+
+  return timerange(from, to)
+    .interval('5m')
+    .rate(1)
+    .generator((timestamp) => {
+      galaxy10.startNewSession();
+      galaxy7.startNewSession();
+      huaweiP2.startNewSession();
+      return [
+        galaxy10
+          .transaction('Start View - View Appearing', 'Android Activity')
+          .timestamp(timestamp)
+          .duration(500)
+          .success()
+          .children(
+            galaxy10
+              .span({
+                spanName: 'onCreate',
+                spanType: 'app',
+                spanSubtype: 'external',
+                'service.target.type': 'http',
+                'span.destination.service.resource': 'external',
```

Comment:
I think we have an `exitSpan` method or something similar for this purpose. (Generally I think we should try to avoid ES field names when using Synthtrace, it should be an implementation detail, so we can more easily rename stuff or create a different output target, like APM Server instead of ES)

Reply:
I'm checking this and I'm not sure which method from Span or MobileDevice, we should use, also there's a `httpSpan()` method in [MobileDevice ](https://github.com/elastic/kibana/blob/ec047aa246f8f17850bf1079db4ff4a6d79a7931/packages/kbn-apm-synthtrace-client/src/lib/apm/mobile_device.ts#L202) that might fit, but then I'm wondering why we have `spanSubtype:external`, shouldn't also be `http`?, I recall we used that field at some point to have the http request but then changed it to `service.target.type`
I didn't add any of these fields, I reused the data generation with sythtrace that we had already for the API tests, so I will need to check with @kpatticha and @AlexanderWert

Reply:
Created [issue](https://github.com/elastic/kibana/issues/155594) to address this